lib/charms/postgresql_k8s/v0/postgresql_tls.py

```diff
@@ -125,8 +125,8 @@ def _on_tls_relation_broken(self, _) -> None:
 
     def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
         """Enable TLS when TLS certificate available."""
-        if event.certificate_signing_request != self.charm.get_secret(SCOPE, "csr"):
-            logger.error("An unknown certificate expiring.")
+        if event.certificate_signing_request.strip() != self.charm.get_secret(SCOPE, "csr").strip():
```

Comment:
The CSR coming from `self.charm.get_secret()` started getting an extra new line at the end. I'm not entirely sure what the root reason for that is.